### PR TITLE
Added RequestTweaker.

### PR DESCRIPTION
**Pull request by yuizumi**
It is more general than and will replace webpackager.Config.RequestHeader.

### CODE REVIEW

File: fetch/request.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+// SetReferer sets the Referer HTTP header with the parent request URL.
+func SetReferer() RequestTweaker {
+	return &setReferer{}
+}
+
+type setReferer struct{}
+
+func (*setReferer) Tweak(req, parent *http.Request) error {
```
Comment (tomokinat) [inline fetch/request.go]: I believe a value receiver is preferred here, per https://github.com/golang/go/wiki/CodeReviewComments#receiver-type.
We may need to modify constructors as well.

File: fetch/request.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+		canonicalKeys[i] = http.CanonicalHeaderKey(key)
+	}
+	return &copyParentHeaders{canonicalKeys}
+}
+
+type copyParentHeaders struct {
+	keys []string
+}
+
+func (cph *copyParentHeaders) Tweak(req, parent *http.Request) error {
```
Comment (tomokinat) [inline fetch/request.go]: Basically same as this is a small struct. (or, `copyParentHeaders` is essentially a `[]string` and `*[]string` receiver is usually only for mutation.)

File: fetch/request.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+// overwritten with the values from the provided http.Header.
+func SetCustomHeaders(header http.Header) RequestTweaker {
+	return &setCustomHeaders{header}
+}
+
+type setCustomHeaders struct {
+	header http.Header
+}
+
+func (sch *setCustomHeaders) Tweak(req, parent *http.Request) error {
```
Comment (tomokinat) [inline fetch/request.go]: Same.

File: fetch/request.go (hunk 1 above)
Comment (yuizumi) [reply fetch/request.go]: I don't think so; I'd apply "when in doubt, use a pointer receiver." If I read the linked doc correctly, there are only a clj;.e  cases where we should use value receivers:

* The receiver is a map, func, or chan
* The receiver is not a map, func, or chan.
* The method doesn't mutate the receiver.
* The receiver doesn't contain anything synchronous.
* The receiver is just a trivial struct, not large.
* There are no changes made to the original receiver.
* The receiver is effectively immutable.
* The receiver isn't analogous to int or str.

File: fetch/request.go (hunk 2 above)
Comment (yuizumi) [reply fetch/request.go]: Same. `copyParentHeaders` consists only of a single slice but it's not a slice by itself. It **isn't** a slice in concept; it just **has** a slice as its extra information. I bet you wouldn't write:

```go
type copyParentHeaders []string
```

File: fetch/request.go (hunk 3 above)
Comment (yuizumi) [reply fetch/request.go]: Same as above.

File: fetch/request.go (hunk 1 above)
Comment (tomokinat) [reply fetch/request.go]: I checked golang's source code and found plenty of both. I'd defer to you, or probably we need to ask some expert, exactly who should have done that kind of reviewing in the first place?

Comment (yuizumi) [reply fetch/request.go]: If we could see both patterns, I'd go with the current code for now ― I've used pointer receivers for all empty structs in this codebase. I'll consider reaching out to some Go expert(s), but AFAICT there's no official channel to receive what you mean by "that kind of reviewing."

Comment (tomokinat) [reply fetch/request.go]: SGTM

**Pull request merged**